### ISSUE TITLE
unable to create records - "create_record: False"

### ISSUE THREAD

**Issue by unclespeedo** [AnalogJ/lexicon#52]
I've installed lexicon with pip in a virtual env with the "[stankevich/python](https://forge.puppet.com/stankevich/python)" puppet module, I only mention this to verify the installation is consistent between hosts.

I'm using the same provider (dnsmadeeasy) and environment variables (LEXICON_DNSMADEEASY_USERNAME, LEXICON_DNSMADEEASY_TOKEN) on both hosts.

When I run try to add records from one host, I'm getting "create_record: False" and when i check the dnsmadeasy web frontend, the record I try to create does not exist.

On the other host, the command returns "create_record: True" and it works as expected.

I'm curious if there are any steps I can take to debug this further.

**Comment by unclespeedo** [comment]
upon further review, the server that isn't working has dns-lexicon 1.1.9 installed, on the servers where it is working, they're running 1.1.7.

**Comment by AnalogJ** [comment]
Hey @unclespeedo 
huh, thats actually really weird, the only changes between 1.1.9 and 1.1.7 are related to `ttl` support and the associated test framework updates

here are the changes: https://github.com/AnalogJ/lexicon/compare/v1.1.7...v1.1.9

can you do a `pip list` and check what other packages are different? (specifically `requests` or `tldextract`)

**Comment by unclespeedo** [comment]
I just reset my environment away and pinned to version 1.1.8 and it works.

FWIW, the working (1.1.7) environment has the following pip list

> distribute (0.7.3)
> dns-lexicon (1.1.7)
> idna (2.1)
> pip (8.1.2)
> requests (2.10.0)
> requests-file (1.4)
> setuptools (25.1.0)
> six (1.10.0)
> tldextract (2.0.1)

I started fresh with 1.1.9 pinned and it failed again (pip list below)

> distribute (0.7.3)
> dns-lexicon (1.1.9)
> idna (2.1)
> pip (8.1.2)
> requests (2.11.1)
> requests-file (1.4)
> setuptools (25.2.0)
> six (1.10.0)
> tldextract (2.0.1)

Since this is fairly easy for me to re-create and get consistent results one way or the other, please let me know if you'd like me to try anything in an effort to resolve the ttl issue with dnsmadeeasy.

**Comment by AnalogJ** [comment]
Thanks for helping me debug this :)

Can you try passing a ttl parameter to lexicon with v.1.1.9? try `--ttl=1800` and if that doesn't work try bumping it up to `--ttl=86400`

I'm thinking that the default value of `300` is too low for dnsmadeeasy and they're throwing an error. It should have been caught by the test suite though, so I'll have to figure out what happened.

**Comment by unclespeedo** [comment]
--ttl=1800 is works successfully

**Comment by AnalogJ** [comment]
Thanks for verifying. So the messed up thing is that the default TTL is 3600 https://github.com/AnalogJ/lexicon/blob/master/lexicon/providers/base.py#L5 

The only reason I tried 1800 and 86400 is that I found them in the DNSMadeEasy Api doc examples. I guess I'll override the default TTL for dnsmadeeasy to 1800 because we know it works. 

And I've figured out why the test suite didn't catch the error, the previous network request recordings weren't deleted. 

I'll get all that fixed and put out a new release (and test suite) this weekend.
